pkg/trace/api/listener.go

```diff
@@ -38,6 +39,7 @@ func NewMeasuredListener(ln net.Listener, name string) net.Listener {
 		timedout: atomic.NewUint32(0),
 		errored:  atomic.NewUint32(0),
 		exit:     make(chan struct{}),
+		sem:      make(chan struct{}, 1000),
```

Comment:
1000 connections should be more than enough to avoid connection refused in most reasonable cases. This just avoids the odd case where a connection explosion could cause unbounded goroutines to be spawned.

The actual limit is imposed in pkg/agent/api/api.go where we deserialize the payloads.